cockpit/src/subscriptions-client.js

```diff
@@ -248,63 +266,72 @@ client.registerSystem = subscriptionDetails => {
-                    }
-                    if (client.config.port !== connection_options.port.v) {
-                        console.debug('saving server.port', connection_options.port.v);
-                        configService.Set('server.port', connection_options.port)
-                            .catch(error => {
-                                console.error('Unable to save candlepin server port', error);
-                            });
-                    }
-                    if (client.config.prefix !== connection_options.handler.v) {
-                        console.debug('saving server.prefix', connection_options.handler.v);
-                        configService.Set('server.prefix', connection_options.handler)
-                            .catch(error => {
-                                console.error('Unable to save candlepin server prefix', error);
-                            });
+                    // Dictionary (key: client.config / rhsm.conf options, value are
+                    // attributes of connection_options) ... ('handler' and 'host' are different)
+                    // Note: only options from [server] section are supported
+                    let dict = {
+                        'hostname': 'host',
+                        'port': 'port',
+                        'prefix': 'handler',
+                        'proxy_hostname': 'proxy_hostname',
+                        'proxy_port': 'proxy_port',
+                        'proxy_user': 'proxy_user',
+                        'proxy_password': 'proxy_password',
+                    };
+                    for (let key in dict) {
+                        // To be sure that config option was loaded from rhsm.conf
+                        if (client.config.hasOwnProperty(key)) {
+                            // Was this config option specified in dialog?
+                            if (connection_options.hasOwnProperty(dict[key])) {
+                                // Is config option in dialog different from  rhsm.conf
+                                if (client.config[key] !== connection_options[dict[key]].v) {
+                                    console.debug('saving: server.' + key, connection_options[dict[key]]);
+                                    configService.Set('server.' + key, connection_options[dict[key]])
+                                        .catch(error => {
+                                            console.error('unable to save server.' + key, error);
+                                        });
+                                }
+                            } else {
+                                // When config option was not specified in dialog, then it is
+                                // necessary to reset it in rhsm.conf (proxy options are optional)
+                                console.debug('resetting: server.' + key);
+                                    configService.Set('server.' + key, {'t': 's', 'v': ''})
```

Comment:
Looks like if you don't specify port or prefix, then they will get reset by this logic. I think the logic should reset to default... you could either pull this values for now from the defaults in `subscriptions-client.js`, or, ideally, we could pull them from the service which would require some enhancement to the d-bus service... I am fine with just pulling from hardcoded values for now with a TODO.

Reply:
@kahowell Thanks for review. Good point. I found another case during testing :-). Please create another Trello card for desired D-Bus changes.